src/org/labkey/remoteapi/assay/Protocol.java

```diff
@@ -0,0 +1,415 @@
+        }
+
+        if (json.containsKey("allowBackgroundUpload"))
+            _allowBackgroundUpload = (Boolean)json.get("allowBackgroundUpload");
+        if (json.containsKey("backgroundUpload"))
+            _backgroundUpload = (Boolean)json.get("backgroundUpload");
+        if (json.containsKey("allowEditableResults"))
+            _allowEditableResults = (Boolean)json.get("allowEditableResults");
+        if (json.containsKey("editableResults"))
+            _editableResults = (Boolean)json.get("editableResults");
+        if (json.containsKey("editableRuns"))
+            _editableRuns = (Boolean)json.get("editableRuns");
+        if (json.containsKey("saveScriptFiles"))
+            _saveScriptFiles = (Boolean)json.get("saveScriptFiles");
+        if (json.containsKey("qcEnabled"))
+            _qcEnabled = (Boolean)json.get("qcEnabled");
+        if (json.containsKey("allowQCStates"))
+            _allowQCStates = (Boolean)json.get("allowQCStates");
+        if (json.containsKey("allowSpacesInPath"))
+            _allowSpacesInPath = (Boolean)json.get("allowSpacesInPath");
+        if (json.containsKey("allowTransformationScript"))
+            _allowTransformationScript = (Boolean)json.get("allowTransformationScript");
+        if (json.containsKey("autoCopyTargetContainerId"))
+            _autoCopyTargetContainerId = (String)json.get("autoCopyTargetContainerId");
+
+        if (json.get("availableDetectionMethods") instanceof JSONArray)
+        {
+            for (Object detectionMethod : (JSONArray)json.get("availableDetectionMethods"))
+                _availableDetectionMethods.add((String)detectionMethod);
+        }
+        if (json.containsKey("selectedDetectionMethod"))
+            _selectedDetectionMethod = (String)json.get("selectedDetectionMethod");
+
+        if (json.containsKey("availableMetadataInputFormats"))
+        {
+            _availableMetadataInputFormats = (HashMap<String,String>)json.get("availableMetadataInputFormats");
+        }
+        if (json.containsKey("selectedMetadataInputFormat"))
+            _selectedMetadataInputFormat = (String)json.get("selectedMetadataInputFormat");
+
+        if (json.get("availablePlateTemplates") instanceof JSONArray)
+        {
+            for (Object plateTemplate : (JSONArray)json.get("availablePlateTemplates"))
+                _availablePlateTemplates.add((String)plateTemplate);
+        }
+        if (json.containsKey("selectedPlateTemplate"))
+            _selectedPlateTemplate = (String)json.get("selectedPlateTemplate");
+
+        if (json.get("protocolTransformScripts") instanceof JSONArray)
+        {
+            for (Object transformScript : (JSONArray)json.get("protocolTransformScripts"))
+                _protocolTransformScripts.add((String)transformScript);
+        }
+        if (json.containsKey("protocolParameters"))
+        {
+            _protocolParameters = (HashMap<String,String>)json.get("protocolParameters");
```

Comment:
Same comment as above for the cast to HashMap, if this is working that's great just want to make sure it is verified.